Adding dependency `com.faster4xml.jackson.dataformat:jackson-dataformat-xml` with:
```patch
diff --git a/build.sbt b/build.sbt
index 344637625f..d6a722598e 100644
--- a/build.sbt
+++ b/build.sbt
@@ -5314,7 +5314,10 @@ lazy val `std-microsoft` = project
       "com.microsoft.sqlserver"   % "mssql-jdbc"              % mssqlserverJDBCVersion,
       "com.azure"                 % "azure-identity"          % azureIdentityVersion exclude ("net.java.dev.jna", "jna") exclude ("net.java.dev.jna", "jna-platform"),
       "com.azure.resourcemanager" % "azure-resourcemanager"   % azureResourceVersion,
-      "com.azure"                 % "azure-storage-blob"      % azureBlobStorageVersion
+      "com.azure"                 % "azure-storage-blob"      % azureBlobStorageVersion,
+      // Using the version of dataformat-xml that azure-core depends on.
+      // So no `jacksonVersion` on purpose.
+      "com.fasterxml.jackson.dataformat" % "jackson-dataformat-xml" % "2.18.1",
     ),
     extractNativeLibs := Def.task {
       import sbt.util.CacheImplicits._
```
does not work. The test fails with:
```
[WARN] [2025-07-02T19:31:57+02:00] [org.enso.librarymanager.local.DefaultLocalLibraryProvider] Candidate library enso_dev.Benchmarks at [***/Benchmarks] may not be AOT ready! Use --jvm option when encoutering problems.
[WARN] [2025-07-02T19:31:57+02:00] [org.enso.librarymanager.local.DefaultLocalLibraryProvider] Candidate library enso_dev.Table_Tests at [***/Table_Tests] may not be AOT ready! Use --jvm option when encoutering problems.
[WARN] [2025-07-02T19:31:57+02:00] [org.enso.librarymanager.local.DefaultLocalLibraryProvider] Candidate library enso_dev.Base_Tests at [***/Base_Tests] may not be AOT ready! Use --jvm option when encoutering problems.
[WARN] [2025-07-02T19:31:57+02:00] [org.enso.librarymanager.local.DefaultLocalLibraryProvider] Candidate library enso_dev.Snowflake_Tests at [***/Snowflake_Tests] may not be AOT ready! Use --jvm option when encoutering problems.
[WARN] [2025-07-02T19:31:57+02:00] [org.enso.librarymanager.local.DefaultLocalLibraryProvider] Candidate library enso_dev.Generic_JDBC_Tests at [***/Generic_JDBC_Tests] may not be AOT ready! Use --jvm option when encoutering problems.
[WARN] [2025-07-02T19:32:02+02:00] [org.enso.microsoft.azure.AzureBlobStorage] List Blob Storage: ensostorage : {}
[ERROR] [2025-07-02T19:32:02+02:00] [com.azure.core.implementation.jackson.ObjectMapperShim] Package versions: jackson-core=2.15.2, jackson-databind=2.15.2, jackson-dataformat-xml=2.18.1, jackson-datatype-jsr310=2.17.2, azure-core=unknown, Troubleshooting version conflicts: https://aka.ms/azsdk/java/dependency/troubleshoot
[FAILED] Azure_Storage: [0/1, 498ms]                                  

    - [FAILED] Connect to dummy blob storage should fail with authentication error [498ms]
        Reason: Expected Authentication error, but actual panic was: (Panic java.lang.LinkageError: Package versions: jackson-core=2.15.2, jackson-databind=2.15.2, jackson-dataformat-xml=2.18.1, jackson-datatype-jsr310=2.17.2, azure-core=unknown, Troubleshooting version conflicts: https://aka.ms/azsdk/java/dependency/troubleshoot java.lang.LinkageError: Package versions: jackson-core=2.15.2, jackson-databind=2.15.2, jackson-dataformat-xml=2.18.1, jackson-datatype-jsr310=2.17.2, azure-core=unknown, Troubleshooting version conflicts: https://aka.ms/azsdk/java/dependency/troubleshoot)Expected False to be True (at Microsoft_Tests/src/Azure_Storage_Spec.enso:16:17-75).


0 tests succeeded.
1 tests failed.
0 tests skipped.
0 groups skipped.

Failed tests: 'Connect.to.dummy.blob.storage.should.fail.with.authentication.error'
```

--- a/build.sbt
+++ b/build.sbt
@@ -3640,11 +3640,14 @@ lazy val `engine-runner` = project
           sys.env.get("CI").isDefined && Platform.isMacOS && Platform.isArm64
         val maxLimit           = if (macArmOnCI) Some(14336) else Some(15608)
         val areStdlibsIncluded = !GraalVM.EnsoLauncher.fast
+        val azureFeature =
+          "org.enso.microsoft.nativeimage.AzureNativeImageFeature"
         val databaseFeature =
           "org.enso.database.nativeimage.SqliteJdbcPatchedFeature"
         val features = Seq(
           "org.enso.interpreter.runtime.nativeimage.NativeLibraryFeature"
-        ) ++ (if (areStdlibsIncluded) Seq(databaseFeature) else Seq())
+        ) ++ (if (areStdlibsIncluded) Seq(databaseFeature, azureFeature)
+              else Seq())
         NativeImage
           .buildNativeImage(
             "enso",

--- a/std-bits/microsoft/src/main/java/org/enso/microsoft/nativeimage/AzureNativeImageFeature.java
+++ b/std-bits/microsoft/src/main/java/org/enso/microsoft/nativeimage/AzureNativeImageFeature.java
@@ -1,0 +1,112 @@
+package org.enso.microsoft.nativeimage;
+
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.jar.JarFile;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeProxyCreation;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+/**
+ * Explicitly registers all the classes that implement {@link com.azure.xml.XmlSerializable}
+ * interface for reflection. This is needed for {@code azure-core} that (de)serializes objects into
+ * XML via reflection.
+ *
+ * <p>Traverses all the {@code azure-*.jar} files on classpath.
+ *
+ * <p>Note that there are various {@code reflection-config.json} files in azure modules, but they do
+ * not list any classes that implement {@code XmlSerializable} interface.
+ */
+public class AzureNativeImageFeature implements Feature {
+
+  private static final String XML_SERIALIZABLE_CLASS_NAME = "com.azure.xml.XmlSerializable";
+
+  @Override
+  public void beforeAnalysis(BeforeAnalysisAccess access) {
+    System.out.println("[AzureNativeImageFeature] Registering classes for reflection...");
+    var xmlSerializableClass = access.findClassByName(XML_SERIALIZABLE_CLASS_NAME);
+    if (xmlSerializableClass == null) {
+      throw new IllegalStateException(
+          "XmlSerializable class not found: " + XML_SERIALIZABLE_CLASS_NAME);
+    }
+    var httpExceptionClass =
+        access.findClassByName("com.azure.core.exception.HttpResponseException");
+    if (httpExceptionClass == null) {
+      throw new IllegalStateException(
+          "HttpResponseException class not found: com.azure.core.exception.HttpResponseException");
+    }
+
+    var classesForReflection = new ArrayList<Class<?>>();
+    for (var path : access.getApplicationClassPath()) {
+      var fileName = path.getFileName().toString();
+      if (fileName.startsWith("azure") && fileName.endsWith(".jar")) {
+        var xmlClasses = findImplementationClasses(access, path, xmlSerializableClass);
+        System.out.println(
+            "[AzureNativeImageFeature] Found "
+                + xmlClasses.size()
+                + " classes implementing XmlSerializable in "
+                + fileName);
+        var respExClasses = findImplementationClasses(access, path, httpExceptionClass);
+        System.out.println(
+            "[AzureNativeImageFeature] Found "
+                + respExClasses.size()
+                + " classes implementing HttpResponseException in "
+                + fileName);
+        classesForReflection.addAll(xmlClasses);
+        classesForReflection.addAll(respExClasses);
+      }
+    }
+    System.out.println(
+        "Registering " + classesForReflection.size() + " classes for runtime reflection.");
+    registerForReflection(xmlSerializableClass);
+    registerForReflection(httpExceptionClass);
+    for (var klazz : classesForReflection) {
+      // TODO: Register only `toXml` and `fromXml` methods.
+      registerForReflection(klazz);
+    }
+  }
+
+  private static void registerForReflection(Class<?> clazz) {
+    RuntimeReflection.register(clazz);
+    RuntimeReflection.register(clazz.getConstructors());
+    RuntimeReflection.register(clazz.getMethods());
+    RuntimeReflection.register(clazz.getFields());
+    RuntimeReflection.registerAllConstructors(clazz);
+    RuntimeReflection.registerAllMethods(clazz);
+    RuntimeReflection.registerAllFields(clazz);
+    if (clazz.isInterface()) {
+      RuntimeProxyCreation.register(clazz);
+    }
+  }
+
+  private static List<Class<?>> findImplementationClasses(
+      BeforeAnalysisAccess access, Path jarPath, Class<?> baseClass) {
+    List<Class<?>> xmlSerializableClasses = new ArrayList<>();
+    try (var jarFile = new JarFile(jarPath.toFile())) {
+      var entries = jarFile.entries();
+      while (entries.hasMoreElements()) {
+        var entry = entries.nextElement();
+        var entryName = entry.getName();
+        if (entryName.endsWith(".class")) {
+          var className = entryName.replace('/', '.').replace(".class", "");
+          var klazz = access.findClassByName(className);
+          if (klazz != null && implementsBaseClass(baseClass, klazz)) {
+            xmlSerializableClasses.add(klazz);
+          }
+        }
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to read classes from azure-core jar: " + jarPath, e);
+    }
+    return xmlSerializableClasses;
+  }
+
+  private static boolean implementsBaseClass(Class<?> baseClass, Class<?> clazz) {
+    return baseClass.isAssignableFrom(clazz)
+        && !clazz.isInterface()
+        && !Modifier.isAbstract(clazz.getModifiers());
+  }
+}

--- a/std-bits/microsoft/src/main/java/org/enso/microsoft/nativeimage/AzureNativeImageFeature.java
+++ b/std-bits/microsoft/src/main/java/org/enso/microsoft/nativeimage/AzureNativeImageFeature.java
@@ -1,5 +1,6 @@
 package org.enso.microsoft.nativeimage;
 
+import com.azure.core.implementation.ReflectionSerializable;
 import java.io.IOException;
 import java.lang.reflect.Modifier;
 import java.nio.file.Path;
@@ -11,11 +12,8 @@ import org.graalvm.nativeimage.hosted.RuntimeProxyCreation;
 import org.graalvm.nativeimage.hosted.RuntimeReflection;
 
 /**
- * Explicitly registers all the classes that implement {@link com.azure.xml.XmlSerializable}
- * interface for reflection. This is needed for {@code azure-core} that (de)serializes objects into
- * XML via reflection.
- *
- * <p>Traverses all the {@code azure-*.jar} files on classpath.
+ * Manually registers some classes and methods for reflection from {@code azure-*.jar} files on
+ * classpath that are not registered by default in {@code reflection-config.json} files.
  *
  * <p>Note that there are various {@code reflection-config.json} files in azure modules, but they do
  * not list any classes that implement {@code XmlSerializable} interface.
@@ -24,8 +22,17 @@ public class AzureNativeImageFeature implements Feature {
 
   private static final String XML_SERIALIZABLE_CLASS_NAME = "com.azure.xml.XmlSerializable";
 
+  private int registeredClasses;
+  private int registeredMethods;
+  private int registeredFields;
+  private int registeredConstructors;
+
   @Override
   public void beforeAnalysis(BeforeAnalysisAccess access) {
+    registeredClasses = 0;
+    registeredMethods = 0;
+    registeredFields = 0;
+    registeredConstructors = 0;
     System.out.println("[AzureNativeImageFeature] Registering classes for reflection...");
     var xmlSerializableClass = access.findClassByName(XML_SERIALIZABLE_CLASS_NAME);
     if (xmlSerializableClass == null) {
@@ -39,41 +46,83 @@ public class AzureNativeImageFeature implements Feature {
           "HttpResponseException class not found: com.azure.core.exception.HttpResponseException");
     }
 
-    var classesForReflection = new ArrayList<Class<?>>();
     for (var path : access.getApplicationClassPath()) {
       var fileName = path.getFileName().toString();
       if (fileName.startsWith("azure") && fileName.endsWith(".jar")) {
         var xmlClasses = findImplementationClasses(access, path, xmlSerializableClass);
-        System.out.println(
-            "[AzureNativeImageFeature] Found "
-                + xmlClasses.size()
-                + " classes implementing XmlSerializable in "
-                + fileName);
+        if (!xmlClasses.isEmpty()) {
+          System.out.println(
+              "[AzureNativeImageFeature] Found "
+                  + xmlClasses.size()
+                  + " classes implementing "
+                  + xmlSerializableClass.getName()
+                  + " in "
+                  + fileName);
+          for (var xmlClass : xmlClasses) {
+            registerXmlClassForReflection(xmlClass);
+          }
+        }
+
         var respExClasses = findImplementationClasses(access, path, httpExceptionClass);
-        System.out.println(
-            "[AzureNativeImageFeature] Found "
-                + respExClasses.size()
-                + " classes implementing HttpResponseException in "
-                + fileName);
-        classesForReflection.addAll(xmlClasses);
-        classesForReflection.addAll(respExClasses);
+        if (!respExClasses.isEmpty()) {
+          System.out.println(
+              "[AzureNativeImageFeature] Found "
+                  + respExClasses.size()
+                  + " classes implementing "
+                  + httpExceptionClass.getName()
+                  + " in "
+                  + fileName);
+          for (var respExClass : respExClasses) {
+            registerForReflection(respExClass);
+          }
+        }
       }
     }
-    System.out.println(
-        "Registering " + classesForReflection.size() + " classes for runtime reflection.");
     registerForReflection(xmlSerializableClass);
     registerForReflection(httpExceptionClass);
-    for (var klazz : classesForReflection) {
-      // TODO: Register only `toXml` and `fromXml` methods.
-      registerForReflection(klazz);
+    System.out.printf(
+        "[AzureNativeImageFeature] Registered %d classes, %d methods, %d fields, and %d"
+            + " constructors for reflection.%n",
+        registeredClasses, registeredMethods, registeredFields, registeredConstructors);
+  }
+
+  private void registerXmlClassForReflection(Class<?> clazz) {
+    RuntimeReflection.register(clazz);
+    registeredClasses++;
+    registerXmlMethodsForReflection(clazz);
+  }
+
+  /**
+   * Checks for the {@code toXml} and {@code fromXml} methods are inspired by {@link
+   * ReflectionSerializable#supportsXmlSerializable(Class)}.
+   */
+  private void registerXmlMethodsForReflection(Class<?> clazz) {
+    for (var method : clazz.getDeclaredMethods()) {
+      if (method.getName().equals("fromXml")) {
+        RuntimeReflection.register(method);
+        registeredMethods++;
+      } else if (method.getName().equals("toXml")) {
+        RuntimeReflection.register(method);
+        registeredMethods++;
+      }
     }
   }
 
-  private static void registerForReflection(Class<?> clazz) {
+  private void registerForReflection(Class<?> clazz) {
     RuntimeReflection.register(clazz);
-    RuntimeReflection.register(clazz.getConstructors());
-    RuntimeReflection.register(clazz.getMethods());
-    RuntimeReflection.register(clazz.getFields());
+    registeredClasses++;
+    var ctors = clazz.getConstructors();
+    RuntimeReflection.register(ctors);
+    registeredConstructors += ctors.length;
+
+    var methods = clazz.getMethods();
+    RuntimeReflection.register(methods);
+    registeredMethods += methods.length;
+
+    var fields = clazz.getFields();
+    RuntimeReflection.register(fields);
+    registeredFields += fields.length;
+
     RuntimeReflection.registerAllConstructors(clazz);
     RuntimeReflection.registerAllMethods(clazz);
     RuntimeReflection.registerAllFields(clazz);

--- a/test/Microsoft_Tests/src/Main.enso
+++ b/test/Microsoft_Tests/src/Main.enso
@@ -3,9 +3,11 @@ from Standard.Base import all
 from Standard.Test import Test
 
 import project.SQLServer_Spec
+import project.Azure_Storage_Spec
 
 main filter=Nothing =
     suite = Test.build suite_builder->
         SQLServer_Spec.add_specs suite_builder
+        Azure_Storage_Spec.add_specs suite_builder
 
     suite.run_with_filter filter

